### PR TITLE
refactor: share OpenAI Chat Completions result materialization

### LINKED ISSUE #847: V1: Consolidate OpenAI Chat Completions result materialization
Parent tracker: #829
Roadmap: [ReqLLM 1.x roadmap](https://github.com/agentjido/req_llm/blob/main/ROADMAP_V1.md) (pending merge)

## Outcome

Consolidate OpenAI Chat Completions buffered, streaming, fixture replay, and `StreamResponse.to_response/1` materialization around the existing accumulator and response-builder contracts.

## Compatibility

**V1 internal — public responses, StreamChunks, fixtures, errors, usage, and finish reasons remain semantically equivalent.**

## Scope

- Reuse and consolidate `ReqLLM.Provider.ChunkAccumulator`, `ReqLLM.Provider.ResponseBuilder`, and `ReqLLM.Response.Stream`.
- Make equivalent buffered and streamed Chat Completions data flow through one final materialization contract.
- Preserve raw provider metadata and all legacy projections.
- Define deterministic terminal error, early cancellation, usage, finish reason, tool-call, and response-ID behavior.
- Remove superseded duplicate accumulation code after parity is proven.

## Out of scope

- A new event accumulator or response-builder hierarchy.
- OpenAI Responses or Anthropic migrations.
- Request planning/encoding or provider transport seam changes.
- Removing legacy Response fields or StreamChunk.

## Acceptance criteria

- [ ] Equivalent buffered and streamed fixtures produce semantically equal results.
- [ ] Existing helpers and stream projections remain equal.
- [ ] Usage, finish reason, identifiers, warnings, tool calls, and errors materialize once.
- [ ] Replay and live decoding share the same finalization contract.
- [ ] No request fixture changes.

## Validation

- [ ] Run Chat Completions buffered/streaming parity tests.
- [ ] Run core, reasoning, tools, structured-output, usage, cancellation, and error replays.
- [ ] Verify fixture and request parity.
- [ ] Run mix test and mix quality.

## Pull request boundary

This issue closes through exactly one pull request containing the implementation, focused tests, and directly affected documentation. Unrelated provider migrations and cleanup are excluded.

Proposed squash title: `refactor: share OpenAI Chat Completions result materialization`

## Dependencies

Depends on #839, #845, and #846.

### PR DESCRIPTION
**Pull request by mikehostetler**
Closes #847

## Summary

- route buffered OpenAI Chat Completions decoding through the existing chunk accumulator and response builder
- share StreamResponse terminal-error and cancellation materialization across helper paths
- preserve buffered response shapes with a narrow compatibility profile for content boundaries, tool arguments, message metadata, wire model IDs, and structured-object behavior
- add contract coverage for buffered/streamed parity, replay projections, errors, cancellation, usage, finish reasons, reasoning, tool calls, logprobs, warnings, and response IDs

## Backward compatibility

- no request encoding or transport changes
- no public struct or callback changes
- no Responses API or provider-specific migrations
- existing buffered serialization and tool-argument edge cases remain locked by regression tests
- `StreamResponse.to_response/1` now returns the same terminal metadata error as `process_stream/2`, matching its documented return contract

## Validation

- `mix test test/req_llm/provider/openai_chat_materialization_test.exs test/req_llm/provider/defaults_test.exs test/req_llm/provider/chunk_accumulator_test.exs test/req_llm/response/stream_test.exs test/req_llm/stream_response_test.exs` (215 passed)
- expanded OpenAI Chat/OpenAI-compatible provider suite (228 passed, 8 excluded)
- `mix test` (3,758 passed, 11 skipped, 145 excluded)
- `mix quality`
- `mix docs` (passes with one pre-existing hidden-module reference warning)

### CODE REVIEW

**Comment by mikehostetler** [comment]
@codex review\n\nPlease focus on backward-compatibility drift in buffered OpenAI-compatible response shapes, tool-call argument edge cases, usage/finish metadata, and StreamResponse terminal error/cancellation semantics.

File: lib/req_llm/provider/defaults/response_builder.ex [hunk 1]
```diff
@@ -43,32 +43,51 @@ defmodule ReqLLM.Provider.Defaults.ResponseBuilder do
...
       tool_calls: if(normalized_tool_calls != [], do: normalized_tool_calls),
-      metadata: build_message_metadata(metadata),
+      metadata: materialize_message_metadata(profile, metadata),
       reasoning_details: reasoning_details
     }
 
-    object = extract_object_from_message(message)
+    object = materialize_object(profile, message)
     usage = normalize_usage_fields(metadata[:usage])
     finish_reason = normalize_finish_reason(metadata[:finish_reason])
```
[the remark this answers is not in the document]
Comment (mikehostetler) [reply lib/req_llm/provider/defaults/response_builder.ex:92]: Verified the buffered decoder still calls `parse_openai_finish_reason/1` before constructing builder metadata, so the builder receives the atom `:error` for an unknown wire value such as `"tool_call"`; `normalize_finish_reason/1` preserves atom values. The old and new paths therefore agree. Commit 271b3e1b adds a direct regression assertion for this Azure/DeepSeek-shaped case, and both the focused suite (215 passed) and full suite (3,758 passed) are green.
